build.py

```diff
@@ -123,33 +124,28 @@ def get_package_filename(p):
     else:
         d = get_distribution_name()
         package_filename_template = 'irods-externals-{0}{1}-{2}_1.0~{5}_{4}.{3}'
-    f = package_filename_template.format(p, v['version_string'], v['consortium_build_number'], t, a, d)
-    return f
+    return package_filename_template.format(p, v['version_string'], v['consortium_build_number'], t, a, d)
 
 def get_versions():
-    with open(script_path+'/versions.json','r') as f:
+    with open(script_path + '/versions.json', 'r') as f:
         return json.load(f)
 
 def get_package_arch():
-    a = platform.uname()[4]
-    if get_package_type() == 'deb' and a == 'x86_64':
+    machine_type = platform.machine()
+    if get_package_type() == 'deb' and machine_type == 'x86_64':
         return 'amd64'
-    return a
+    return machine_type
 
 def get_package_type():
     log = logging.getLogger(__name__)
-    pld = platform.linux_distribution()[0]
-    if pld == '':
-        import distro
-        pld = distro.linux_distribution()[0]
-
-    log.debug('linux distribution detected: {0}'.format(pld))
-    if pld in ['debian', 'Ubuntu']:
+    distro_id = distro.id()
+    log.debug('linux distribution detected: {0}'.format(distro_id))
+    if distro_id in ['debian', 'ubuntu']:
         pt = 'deb'
-    elif pld in ['Rocky Linux', 'AlmaLinux', 'CentOS', 'CentOS Linux', 'Red Hat Enterprise Linux Server', 'Scientific Linux', 'openSUSE ', 'openSUSE Leap', 'SUSE Linux Enterprise Server', 'SLES']:
+    elif distro_id in ['rocky', 'almalinux', 'centos', 'rhel', 'scientific', 'opensuse', 'sles']:
```

Comment:
notice that the list is smaller.

i removed some entries because they seemed redundant. i have not verified if the new list covers all entries from the original list.

Reply:
i think we're probably good here - the different ages of CentOS and SUSE reported themselves differently.  If we've got the current ones covered, that's enough.  Can always add more.

Reply:
agreed